python-sdk/tests/test_example_dags.py

```diff
@@ -93,3 +93,22 @@ def test_example_dynamic_task_map_dag(session, dag_id):
     if dag is None:
         raise NameError(f"The DAG with dag_id: {dag_id} was not found")
     wrapper_run_dag(dag)
+
+
+@pytest.mark.skipif(
+    airflow.__version__ < "2.4.0", reason="Require Airflow version >= 2.4.0"
+)
+@pytest.mark.parametrize(
+    "dag_id",
+    [
+        "example_dataset_producer",
+    ],
+)
+def test_example_dataset_dag(session, dag_id):
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    db = DagBag(dir_path + "/../example_dags")
+    dag = db.get_dag(dag_id)
+
+    if dag is None:
+        raise NameError(f"The DAG with dag_id: {dag_id} was not found")
+    wrapper_run_dag(dag)
```

Comment:
Did you test it with 2.4.0b1 .. you can test it out on CI by bumping airflow dep to >=2.4.0b1 

I have a feeling this won't work as the second Dataset wont be triggered by the logic in `wrapper_run_dag`

Reply:
Bump the following to 2.4.0b1
https://github.com/astronomer/astro-sdk/blob/330c02a4d455759f9a164dde253a32747a7d2f5f/python-sdk/pyproject.toml#L20

Reply:
No I havent tested with 2.4.0b1, doing it now

Reply:
somehow CI doesnt seem to run tests on 2.4.0b1

Reply:
It has picked up the DAG successfully
`5.85s call     tests/test_example_dags.py::test_example_dataset_dag[example_dataset_producer]`

Reply:
As discussed, moved the test to `python-sdk/tests/airflow/test_datasets.py` . Its working fine with `2.4.0b1`